punq/__init__.py

```diff
@@ -208,19 +224,27 @@ def _update_localns(self, service):
         else:
             self._localns[service] = service
 
-    def register(self, service, factory=empty, instance=empty, **kwargs):
+    def register(self, service, factory=empty, instance=empty, scope=empty, **kwargs):
         resolve_args = kwargs or {}
 
         if instance is not empty:
+            if scope not in [Scope.instance, empty]:
+                raise InvalidRegistrationException(
+                    f"Must use instance scope to register instances, not {scope}"
+                )
             self.register_service_and_instance(service, instance)
-        elif factory is empty:
-            self.register_concrete_service(service)
-        elif callable(factory):
-            self.register_service_and_impl(service, factory, resolve_args)
         else:
-            raise InvalidRegistrationException(
-                f"Expected a callable factory for the service {service} but received {factory}"
-            )
+            scope = scope if scope != empty else self._default_scope
```

Comment:
If this declaration were moved to the top of the function, could we change line 231 `if scope not in [Scope.instance, empty]` for `if scope is not Scope.instance` ?